src/Bindings/ManualBindings_BlockArea.cpp

```diff
@@ -951,6 +951,114 @@ static int SetRelBlock(lua_State * a_LuaState)
 
 
 
+static int tolua_cBlockArea_SetBlockTypeMeta(lua_State * a_LuaState)
+{
+	// Check the common params:
+	cLuaState L(a_LuaState);
+	if (!L.CheckParamSelf("cBlockArea"))
+	{
+		return 0;
+	}
+
+	// Read the common params:
+	cBlockArea * self;
+	if (!L.GetStackValues(1, self))
+	{
+		return cManualBindings::ApiParamError(a_LuaState, "Cannot read the 'self' param.");
+	}
+
+	// Check the datatype's presence:
+	if ((self->GetDataTypes() & cBlockArea::baTypes) == 0)
+	{
+		return cManualBindings::ApiParamError(a_LuaState, "The area doesn't contain the datatype (%d).", cBlockArea::baTypes);
+	}
+
+	// Check the datatype's presence:
+	if ((self->GetDataTypes() & cBlockArea::baMetas) == 0)
+	{
+		return cManualBindings::ApiParamError(a_LuaState, "The area doesn't contain the datatype (%d).", cBlockArea::baMetas);
+	}
+
+	// Read the overloaded params:
+	Vector3i coords;
+	auto idx = readVector3iOverloadParams(L, 2, coords, "coords");
+	if (!self->IsValidCoords(coords))
+	{
+		return cManualBindings::ApiParamError(a_LuaState, "The coords ({%d, %d, %d}) are out of range ({%d, %d, %d}).",
+			coords.x, coords.y, coords.z,
+			self->GetSizeX(), self->GetSizeY(), self->GetSizeZ()
+		);
+	}
+
+	BLOCKTYPE Block;
+	L.GetStackValues(idx, Block);
+
+	NIBBLETYPE Meta;
+	L.GetStackValues(idx + 1, Meta);
```

Comment:
Read both `block` and `meta` at the same time, using `GetStackValues`; prefer lowerCamelCase.